### PR TITLE
CORE-7116 : UTXO Ledger Contract Verification

### PR DESCRIPTION
**Pull request by MrMatthewLayton**
Added exception APIs for contract verification.

### CODE REVIEW

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractVerificationFailureReason.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * Defines a contract verification failure reason.
+ *
+ * @property command The contract [Command] that caused the verification failure.
+ * @property cause The underlying cause of the verification failure.
+ */
+@CordaSerializable
+@DoNotImplement
+interface ContractVerificationFailureReason {
```
Comment (lankydan) [inline ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractVerificationFailureReason.kt]: Small thing, does the "reason" add anything here or is `ContractVerificationFailure` enough?

Comment (MrMatthewLayton) [reply ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractVerificationFailureReason.kt]: Yes, on balance, I think `ContractVerificationFailure` is enough.

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractVerificationFailureReason.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    val command: Class<out Command>
+    val cause: Exception?
```
Comment (mnesbit) [inline ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractVerificationFailureReason.kt]: Won't work!

**Review by blsemo — DISMISSED** [review]
Looks fair enough to me.

**Comment by MrMatthewLayton** [comment]
@vkolomeyko This was merged in a separate branch, so closing this one.

**Pull request closed without merging**